UI/Entities/Game.cs

```diff
@@ -5,102 +5,123 @@ namespace UI.Entities;
+    {
+        return Guest is not null;
+    }
 
-  public void Move(Move move)
-  {
-    if (Host != null && Host.ConnectionId == move.ConnectionId)
+    public void Move(Move move)
     {
-      Host.HasTurn = false;
-      if (Guest != null) Guest.HasTurn = true;
+        if (Host != null && Host.ConnectionId == move.ConnectionId)
+        {
+            Host.HasTurn = false;
+            if (Guest != null) Guest.HasTurn = true;
+        }
+
+        if (Guest != null && Guest.ConnectionId == move.ConnectionId)
+        {
+            Guest.HasTurn = false;
+            if (Host != null) Host.HasTurn = true;
+        }
+
+        _gameBoard[move.Index] = move.Mark;
     }
 
-    if (Guest != null && Guest.ConnectionId == move.ConnectionId)
+    public void RemovePlayer(string connectionId)
     {
-      Guest.HasTurn = false;
-      if (Host != null) Host.HasTurn = true;
+        if (Host is not null)
+            if (Host.ConnectionId == connectionId)
+                Host = null;
+
+        if (Guest is null) return;
+        if (Guest.ConnectionId == connectionId)
+            Guest = null;
     }
 
-    _gameBoard[move.Index] = move.Mark;
-  }
+    public GameOutcome? HasOutcome(Marks mark)
+    {
+        bool win = HasWinningRow(mark) || HasWinningColumn(mark) || HasDiagonalWon(mark);
+
+        return win switch
+        {
+            false when !CanContinue() => GameOutcome.Draw,
+            true => GameOutcome.Win,
+            _ => null
+        };
+    }
 
-  public void RemovePlayer(string connectionId)
-  {
-    if (Host is not null)
-      if (Host.ConnectionId == connectionId)
-        Host = null;
+    private bool CanContinue()
+    {
+        return _gameBoard.Any(x => x == Marks.NotSet);
+    }
 
-    if (Guest is null) return;
-    if (Guest.ConnectionId == connectionId)
-      Guest = null;
-  }
+    private bool HasWinningRow(Marks mark)
+    {
+        return (_gameBoard[0] == mark && _gameBoard[1] == mark && _gameBoard[2] == mark) ||
+               (_gameBoard[3] == mark && _gameBoard[4] == mark && _gameBoard[5] == mark) ||
+               (_gameBoard[6] == mark && _gameBoard[7] == mark && _gameBoard[8] == mark);
+    }
+
+    private bool HasWinningColumn(Marks mark)
+    {
+        return (_gameBoard[0] == mark && _gameBoard[3] == mark && _gameBoard[6] == mark) ||
+               (_gameBoard[1] == mark && _gameBoard[4] == mark && _gameBoard[7] == mark) ||
+               (_gameBoard[2] == mark && _gameBoard[5] == mark && _gameBoard[8] == mark);
+    }
 
-  public GameOutcome? HasOutcome(Marks mark)
-  {
-    bool win = HasWinningRow(mark) || HasWinningColumn(mark) || HasDiagonalWon(mark);
+    private bool HasDiagonalWon(Marks mark)
+    {
+        return (_gameBoard[0] == mark && _gameBoard[4] == mark && _gameBoard[8] == mark) ||
+               (_gameBoard[2] == mark && _gameBoard[4] == mark && _gameBoard[6] == mark);
+    }
 
-    return win switch
+    public void SetWinnerIfAny(Move model)
```

Comment:
This function is unnecessary. The logic you have in it should be incorporated into the `.HasOutcome()` method. As part of checking the game outcome, we're setting the winner and loser.